clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -615,21 +613,13 @@ public static String stackTrace(Throwable e) {
     }
 
     /**
-     * Attempt to read a file as a string
-     * @throws IOException
+     * Read a file as string and return the content. The file is treated as a stream and no seek is performed.
+     * This allows the program to read from a regular file as well as from a pipe/fifo.
      */
-    public static String readFileAsString(String path, Charset charset) throws IOException {
-        if (charset == null) charset = Charset.defaultCharset();
-
-        try (FileChannel fc = FileChannel.open(Paths.get(path))) {
-            MappedByteBuffer bb = fc.map(FileChannel.MapMode.READ_ONLY, 0, fc.size());
-            return charset.decode(bb).toString();
-        }
-
-    }
-
     public static String readFileAsString(String path) throws IOException {
-        return Utils.readFileAsString(path, Charset.defaultCharset());
+        log.debug("Reading file {}", path);
```

Comment:
Hmm, why did we do this? I thought we'd have a try/catch block.

Reply:
The method signature was throwing IOException, and implementation wasn't. The only thing try-catch was adding was to log the file path in case things go wrong, so instead added a log statement for that.